### PR TITLE
Unified design picker: Tailored UI based on intent

### PR DESCRIPTION
**Pull request by bangank36**
#### Proposed Changes

- Update query params to send `site intent` 

#### Testing Instructions

- Navigate to /designSetup step
- Check for the intent parameter in Network tab

Related to #65924

### CODE REVIEW

File: packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts [hunk 1]
```diff
@@ -53,7 +54,6 @@ export function useStarterDesignsQuery(
 		...queryOptions,
 	} );
 }
-
```
Comment (fushar) [inline packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts]: Revert this change (keep the blank line)? 😄

Comment (bangank36) [reply packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts]: My bad, fixed

**Review by arthur791004 — APPROVED** [review]
